xla/service/gpu/gpu_compiler.cc

```diff
@@ -1586,7 +1586,17 @@ absl::Status GpuCompiler::OptimizeHloPostLayoutAssignment(
     return Compiler::TargetConfig{gpu_target_config_proto};
   }
   if (executor) {
-    return Compiler::TargetConfig{executor};
+    Compiler::TargetConfig target_config = Compiler::TargetConfig{executor};
+    int64_t device_memory_size =
+      target_config.device_description.device_memory_size();
+    // Checking for device_memory_size == -1 is how we detect that we are
+    // running on Nvidia's software simulator. When running on simulation,
+    // the config from StreamExecutor is inaccurate, so we must load the
+    // hard-coded config from a file.
+    CHECK_NE(device_memory_size, -1)
```

Comment:
I don't think we want to check-fail, maybe error out with a Status instead? Also maybe put most of the comment message into the returned string?

Reply:
done